core-rust/state-manager/src/mempool/pending_transaction_result_cache.rs

```diff
@@ -510,10 +512,10 @@ impl PendingTransactionResultCache {
         let mut write_mempool = self.mempool.write();
         if attempt.rejection.is_some() {
             write_mempool.remove_by_payload_hash(&notarized_transaction_hash);
-        } else if let Some(state_version) = attempt.against_state.specific_version() {
+        } else if let AtState::Specific(specific_state) = &attempt.against_state {
             write_mempool.update_transaction_executed_state_version(
                 &notarized_transaction_hash,
-                state_version,
+                specific_state.committed_version(),
```

Comment:
Actually, I've just noticed that this code in `state_computer` `prepare` should be removed too, as it's now handled by `track_transaction_result` a few lines below:
```
        self.mempool_manager
            .remove_rejected(&pending_rejected_transactions);
```

...and we could then remove  `mempool_manager.remove_rejected` and `mempool.remove_by_payload_hash`